test/cluster/test_cluster_features.py

```diff
@@ -146,13 +146,13 @@ async def test_joining_old_node_fails(manager: ManagerClient) -> None:
 
     # Try to add a node that doesn't support the feature - should fail
     new_server_info = await manager.server_add(start=False, property_file=servers[0].property_file())
-    await manager.server_start(new_server_info.server_id, expected_error="Feature check failed")
+    await manager.server_start(new_server_info.server_id, expected_error="Feature check failed|received notification of being banned from the cluster from")
```

Comment:
It's not an ideal change because, in theory, the node could be banned for a reason other than not supporting the feature, and the test is specifically about cluster features. We weaken the test a bit here. To prevent this, we could e.g. look for
```cpp
rtlogger.warn("node {} does not understand some features: {}", node.id, unsupported_features);
```
in the topology coordinator's logs. One line should appear here after the bootstrap failure, and another line should appear after the replace failure below. Note that we can also match the expected values of `node.id` and `unsupported_features` in the test, so we don't need to use log marks.